packages/bs-neox/src/__tests__/BlockscoutBDSNeoXFullTransactionsByAddress.spec.ts

```diff
@@ -0,0 +1,299 @@
+          networkFeeAmount: expect.any(String),
+          systemFeeAmount: expect.any(String),
+          events: expect.arrayContaining([
+            expect.objectContaining({
+              eventType: expect.any(String),
+              amount: expect.any(String),
+              methodName: expect.any(String),
+              from: expect.anything(),
+              fromUrl: expect.anything(),
+              to: expect.anything(),
+              toUrl: expect.anything(),
+              hash: expect.any(String),
+              tokenType: 'generic',
+            }),
+          ]),
+        }),
+      ]),
+    })
+  })
+
+  // There isn't any Neo X address with cursor yet (you need to use the pageLimit for test it)
+  it.skip('Should be able to get transactions when send the nextCursor param using Neo X Mainnet network', async () => {
+    const newParams = {
+      ...params,
+      dateFrom: new Date('2024-04-26T12:00:00').toJSON(),
+      dateTo: new Date('2025-04-25T12:00:00').toJSON(),
+      address: '0x1212000000000000000000000000000000000004',
+    }
+
+    const response = await blockscoutBDSNeoX.getFullTransactionsByAddress(newParams)
+    const nextResponse = await blockscoutBDSNeoX.getFullTransactionsByAddress({
+      ...newParams,
+      nextCursor: response.nextCursor,
+    })
+
+    expect(response.nextCursor).toBeTruthy()
+    expect(response.data.length).toBeTruthy()
+    expect(nextResponse.nextCursor).toBeFalsy()
+    expect(nextResponse.data.length).toBeTruthy()
+  }, 60000)
+
+  // The NFTs on Neo X should be implemented in future in Dora (remove mock to test)
+  it.skip('Should be able to get transactions with NFTs when it was called using Neo X Mainnet network', async () => {
+    const response = await blockscoutBDSNeoX.getFullTransactionsByAddress({
+      ...params,
+      address: '0xE3aBC0b2A74FD2eF662b1c25C9769398f53b4304',
+      dateFrom: new Date('2024-01-01T12:00:00').toJSON(),
+      dateTo: new Date('2024-12-31T12:00:00').toJSON(),
+    })
+
+    const nftEvents = response.data
+      .flatMap(({ events }) => events)
+      .filter(({ eventType }) => eventType === 'nft') as FullTransactionNftEvent[]
+
+    expect(nftEvents).toEqual(
+      expect.arrayContaining([
+        expect.objectContaining({
+          eventType: 'nft',
+          amount: expect.any(String),
+          methodName: expect.any(String),
+          from: expect.anything(),
+          fromUrl: expect.anything(),
+          to: expect.anything(),
+          toUrl: expect.anything(),
+          hash: expect.any(String),
+          hashUrl: expect.any(String),
+          tokenId: expect.any(String),
+          tokenType: 'generic',
+          nftImageUrl: 'nftImage',
+          nftUrl: expect.any(String),
+          name: 'nftName',
+          collectionName: 'nftCollectionName',
+        }),
+      ])
+    )
+  }, 30000)
+
+  it('Should be able to export transactions when is using a Neo X Testnet network', async () => {
```

Comment:
I prefer create a describe for each method.

Reply:
same https://github.com/CityOfZion/blockchain-services/pull/173#discussion_r2100394838